src/neoxp/Node/NodeUtility.cs

```diff
@@ -208,6 +211,197 @@ public static void SignOracleResponseTransaction(ProtocolSettings settings, Expr
+                // to avoid having contracts with duplicate id's. This is important because the contract id is part of the
+                // StorageContext used with Storage syscalls and else we'll potentially override storage keys or iterate
+                // over keys that shouldn't exist for one of the contracts.
+                if (state.Id <= LastUsedContractId(snapshot))
+                {
+                    state.Id = GetNextAvailableId(snapshot);
+                }
+                else
+                {
+                    // Update available id such that a regular contract deploy will use the right next id;
+                    SetNextAvailableContractId(snapshot, state.Id + 1);
+                }
+
+                snapshot.Add(key, new StorageItem(state));
+            }
+            else
+            {
+                if (overwriteContract)
+                {
+                    // Note: a ManagementContract.Update() will not change the contract hash. Not even if the NEF changed.
+                    localContract.Nef = state.Nef;
+                    localContract.Manifest = state.Manifest;
+                    localContract.UpdateCounter = state.UpdateCounter;
+                }
+                else
+                {
+                    // check to make sure the existing contract matches the downloaded contract
+                    // except for id, which is allowed to be different from downloaded contract
+                    if (!localContract.Hash.Equals(state.Hash)
+                        || localContract.UpdateCounter != state.UpdateCounter
+                        || !localContract.Nef.ToArray().SequenceEqual(state.Nef.ToArray())
+                        || !localContract.Manifest.ToJson().ToByteArray(false).SequenceEqual(state.Manifest.ToJson().ToByteArray(false)))
+                    {
+                        throw new Exception("Downloaded contract already exists. Use --force to overwrite");
+                    }
+                }
+            }
+
+            var contractId = localContract is null ? state.Id : localContract.Id;
+            if (overwriteStorage)
+            {
+                // the storage schema might have changed therefore we first clear all storage
+                byte[] prefix_key = StorageKey.CreateSearchPrefix(contractId, default);
+                foreach (var (k, v) in snapshot.Find(prefix_key))
+                {
+                    snapshot.Delete(k);
+                }
+
+                for (int i = 0; i < storagePairs.Count; i++)
+                {
+                    snapshot.Add(
+                        new StorageKey { Id = contractId, Key = Convert.FromBase64String(storagePairs[i].key) },
+                        new StorageItem(Convert.FromBase64String(storagePairs[i].value)));
+                }
+            }
+            else
+            {
```

Comment:
If the downloaded contract is new and `force` is `NONE`. It will fail on the `storageCount` mismatch throwing a misleading error that storage already exists while there's actually nothing. There's also no logic to persist the storage of a completely new contract.

Reply:
WIll fix